projects/dsp-ui/src/lib/viewer/values/interval-value/interval-input/interval-input.component.ts

```diff
@@ -27,6 +27,20 @@ export class IntervalInputErrorStateMatcher implements ErrorStateMatcher {
     }
 }
 
+/** Interval must have a start and end of the same type, either both numbers or both null */
+export function startEndSameTypeValidator(otherInterval: FormControl): ValidatorFn {
+    return (control: AbstractControl): { [key: string]: any } | null => {
+
+        let invalid = true;
+        if (typeof(control.value) === 'number' && typeof(otherInterval.value) === 'number' ||
```

Comment:
Couldn't you just check for `null`?

Something like `invalid = !(control.value === null && otherInterval.value === null || control.value !== null && otherInterval.value !== null)` 

 I think the `input` only allows for numeric values.

Reply:
This was me being overly paranoid. Refactored in 3c0a216